iothub/service/src/AmqpTransportLog.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices
+{
+    // This class overrides the trace providers available in Azure.Amqp library to
+    // write into the service client log provider as well.
+    internal class AmqpTransportLog : AmqpTrace
+    {
+        public override void AmqpAbortThrowingException(string exception)
+        {
+            if (Logging.IsEnabled)
+            {
+                Logging.Info(exception, "AmqpAbortThrowingException:", "AmqpTransportLog");
+            }
+
+            base.AmqpAbortThrowingException(exception);
+        }
+
+        public override void AmqpAddSession(object source, object session, ushort localChannel, ushort remoteChannel)
+        {
+            if (Logging.IsEnabled)
+            {
+                Logging.Associate(source, session, "AmqpAddSession:", $"{" source:"}{source}.{" session:"}{session}.{" localChannel:"}.{localChannel}.{" remoteChannel:"}.{remoteChannel}");
```

Comment:
What does this string interpolation mean:
`{" source:"}`

Reply:
yeah, that is weird, I'll fix it.